compiler/plugins/target/ROCM/ROCMTarget.cpp

```diff
@@ -51,13 +51,20 @@ namespace mlir::iree_compiler::IREE::HAL {
 
 namespace {
 
-struct ROCMOptions {
+class ROCMOptions {
+public:
   std::string targetChip = "gfx908";
-  bool linkBitcode = false;
-  std::string bitcodeDirectory;
+  bool linkBitcode = true;
   int wavesPerEu = 0;
   std::string enableROCMUkernels = "none";
 
+  const std::string &getBitcodeDirectory() const {
+    if (bitcodeDirectory.empty()) {
+      return getDefaultBitcodeDirectory();
+    }
+    return bitcodeDirectory;
+  }
```

Comment:
Also, /facepalm at how I had this written before... thanks for the function + private variable change here.

For LLVMCPUTarget, I put a helper class between the options and what the target actually sees: https://github.com/openxla/iree/blob/d05b4a16f826ca328f7e737ae473e2b581dc950f/compiler/plugins/target/LLVMCPU/LLVMTargetOptions.cpp#L656-L693

I guess that could be folded down to a single class by using a similar technique (private vars for all the internals, public functions to get values dependent on extra code)

Reply:
Sometimes it takes a python programmer to properly encapsulate code.